driver/terraform.go

```diff
@@ -65,71 +65,52 @@ type TerraformConfig struct {
-// installed to the configured path.
-func (tf *Terraform) Init(ctx context.Context) error {
-	if _, err := os.Stat(tf.workingDir); os.IsNotExist(err) {
-		if err := os.MkdirAll(tf.workingDir, workingDirPerms); err != nil {
-			log.Printf("[ERR] (driver.terraform) error creating base work directory: %s", err)
-			return err
+func NewTerraform(config *TerraformConfig) (*Terraform, error) {
+	if _, err := os.Stat(config.WorkingDir); os.IsNotExist(err) {
+		if err := os.Mkdir(config.WorkingDir, workingDirPerms); err != nil {
+			log.Printf("[ERR] (driver.terraform) error creating task work directory: %s", err)
+			return nil, err
 		}
 	}
 
-	if isTFInstalled(tf.path) {
-		tfVersion, compatible, err := isTFCompatible(ctx, tf.workingDir, tf.path)
-		if err != nil {
-			if strings.Contains(err.Error(), "exec format error") {
-				return errIncompatibleTerraformBinary
-			}
-			return err
-		}
-
-		if !compatible {
-			return errUnsupportedTerraformVersion
-		}
-
-		tf.version = tfVersion.String()
-		log.Printf("[INFO] (driver.terraform) skipping install, terraform %s "+
-			"already exists at path %s/terraform", tf.version, tf.path)
-		return nil
+	tfClient, err := newClient(&clientConfig{
+		task:       config.Task,
+		clientType: config.ClientType,
+		log:        config.Log,
+		persistLog: config.PersistLog,
+		path:       config.Path,
+		workingDir: config.WorkingDir,
+	})
+	if err != nil {
+		log.Printf("[ERR] (driver.terraform) init client type '%s' error: %s", config.ClientType, err)
+		return nil, err
 	}
 
-	log.Printf("[INFO] (driver.terraform) installing terraform to path '%s'", tf.path)
-	if err := tf.install(ctx); err != nil {
-		log.Printf("[ERR] (driver.terraform) error installing terraform: %s", err)
-		return err
+	handler, err := getTerraformHandlers(config.Task)
+	if err != nil {
+		return nil, err
 	}
-	log.Printf("[INFO] (driver.terraform) successfully installed terraform")
-	return nil
+
+	return &Terraform{
+		task:              config.Task,
+		backend:           config.Backend,
+		requiredProviders: config.RequiredProviders,
+		workingDir:        config.WorkingDir,
+		worker:            newWorker(defaultRetry),
+		client:            tfClient,
+		postApply:         handler,
+	}, nil
 }
 
 // Version returns the Terraform CLI version for the Terraform driver.
 func (tf *Terraform) Version() string {
-	return tf.version
+	return TerraformVersion
 }
 
 // InitTask initializes the task by creating the Terraform root module, creating
 // client to execute task, and retrieving post-Terraform apply handlers
-func (tf *Terraform) InitTask(task Task, force bool) error {
-	modulePath := filepath.Join(tf.workingDir, task.Name)
-	if _, err := os.Stat(modulePath); os.IsNotExist(err) {
-		if err := os.Mkdir(modulePath, workingDirPerms); err != nil {
-			log.Printf("[ERR] (driver.terraform) error creating task work directory: %s", err)
-			return err
-		}
-	}
+func (tf *Terraform) InitTask(ctx context.Context, force bool) error {
```

Comment:
I'm not seeing context being consumed. I might have missed it. is it being used or plans for later?

Reply:
yeah it's not being used. i was iffy on whether to keep the function signature consistent across the driver methods (Inspect and Apply both have ctx). Would you prefer this to be removed until used?

Reply:
ah I see. I personally lean towards removing until use but I also see what you're saying about consistency. I'm ok either way!

Reply:
Ok, will remove. `InitTask` in nature is different than the others that are longer running functions and has more of a consisten/predictable execution time.

Reply:
I also added context handling in the `baseController.init(ctx)` to stop looping through tasks early if needed.